### PR TITLE
DAOS-11599 client: Update test to run leak checking on mwc tests.

### PR DESCRIPTION
**Pull request by ashleypittman**
Required-githooks: true

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
@mchaarawi there are only actually three errors with this PR which is fewer than I expected.  We could in theory land as-is and the warnings-ng plugin should use the correct reference builds and not cause any problems with PRs but that hasn't been our experiences in the past so I'll tidy up these error messages and add suppressions for them to the code where they're passed.

**Comment by mchaarawi** [comment]
> @mchaarawi there are only actually three errors with this PR which is fewer than I expected. We could in theory land as-is and the warnings-ng plugin should use the correct reference builds and not cause any problems with PRs but that hasn't been our experiences in the past so I'll tidy up these error messages and add suppressions for them to the code where they're passed.

we don't want to add suppressions for those. they are legitimate errors that need to be flagged outside of this test. i think this is a case where you should just leave it alone :-)

**Comment by ashleypittman** [comment]
I checked all the dfuse and dfs using of stderror as a result of this so the code should be a lot more consistent now.

File: src/client/dfuse/dfuse_main.c [hunk 1]
```diff
@@ -557,8 +557,8 @@ main(int argc, char **argv)
 	 */
 	duns_attr.da_flags = DUNS_NO_REVERSE_LOOKUP;
 	rc = duns_resolve_path(dfuse_info->di_mountpoint, &duns_attr);
-	DFUSE_TRA_INFO(dfuse_info, "duns_resolve_path() on mountpoint returned %d %s",
-		       rc, strerror(rc));
+	DFUSE_TRA_INFO(dfuse_info, "duns_resolve_path() on mountpoint failed: %d (%s)", rc,
```
Comment (daltonbohning) [inline src/client/dfuse/dfuse_main.c:560]: This isn't necessarily a failure, right?

File: src/tests/ftest/cart/util/cart_logtest.py [hunk 2]
```diff
@@ -418,11 +419,16 @@ def _check_pid_from_log_file(self, pid, abort_on_warning, leak_wf, show_memleaks
...
-                        if line.level < cart_logparse.LOG_LEVELS['ERR']:
-                            show_line(line, 'HIGH', 'error in strict mode')
-                        else:
-                            show_line(line, 'NORMAL', 'warning in strict mode')
-                        warnings_mode = True
+                        show_this = True
+                        for ignores in self.skip_suffixes:
+                            if line.get_msg().endswith(ignores):
+                                show_this = False
+                        if show_this:
```
Comment (daltonbohning) [inline src/tests/ftest/cart/util/cart_logtest.py:426]: FYI - for a one-liner that also early-exits (`map` returns a generator)
```suggestion
                        if not any(map(line.get_msg().endswith, self.skip_suffixes)):
```

**Review by daltonbohning — APPROVED** [review]

File: src/client/dfuse/dfuse_main.c (hunk 1 above)
Comment (ashleypittman) [reply src/client/dfuse/dfuse_main.c:560]: True, in fact I've just seen this is a INFO message not an ERROR.  I still think the change is valid however, just for consistency.

Comment (mchaarawi) [reply src/client/dfuse/dfuse_main.c:560]: how is it valid that a return value of 0 would log that it failed?

File: src/client/dfuse/ops/lookup.c [hunk 3]
```diff
@@ -207,13 +207,13 @@ check_for_uns_ep(struct dfuse_projection_info *fs_handle,
 
 	rc = dfs_release(ie->ie_obj);
 	if (rc) {
-		DFUSE_TRA_ERROR(dfs, "dfs_release() failed: (%s)", strerror(rc));
+		DFUSE_TRA_ERROR(dfs, "dfs_release() returned: %d (%s)", rc, strerror(rc));
```
Comment (mchaarawi) [inline src/client/dfuse/ops/lookup.c:210]: here it's the opposite. failed is more appropriate than returned.

File: src/client/dfuse/dfuse_main.c (hunk 1 above)
Comment (ashleypittman) [reply src/client/dfuse/dfuse_main.c]: Sorry, I didn't look closely enough at the change in light of Daltons comment.  Consistency with use of strerror is good but you're correct I should not have changed this to failed, I've reverted that part of the change.

File: src/client/dfuse/ops/lookup.c (hunk 3 above)
Comment (ashleypittman) [reply src/client/dfuse/ops/lookup.c]: I've reverted this working change as well.

File: src/tests/ftest/cart/util/cart_logtest.py (hunk 2 above)
Comment (ashleypittman) [reply src/tests/ftest/cart/util/cart_logtest.py:426]: This PR is going to conflict with https://github.com/daos-stack/daos/pull/12112 whatever I do here so I'll leave this for now and fix up once that has landed.

**Review by mchaarawi — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**